test/espnet2/asr/decoder/test_transformer_decoder.py

```diff
@@ -26,14 +50,132 @@ def test_TransformerDecoder_backward(input_layer, normalize_before, use_output_l
     z_all.sum().backward()
 
 
-def test_TransformerDecoder_init_state():
-    decoder = TransformerDecoder(10, 12)
+@pytest.mark.parametrize(
+    "decoder_class",
+    [
+        TransformerDecoder,
+        LightweightConvolutionTransformerDecoder,
+        LightweightConvolution2DTransformerDecoder,
+        DynamicConvolutionTransformerDecoder,
+        DynamicConvolution2DTransformerDecoder,
+    ],
+)
+def test_TransformerDecoder_init_state(decoder_class):
+    decoder = decoder_class(10, 12)
```

Comment:
I'm afraid that this model is too large for unit tests. For example, default `linear_units` is 2048. That should be smaller.

Reply:
I see